ravenframework/CodeInterfaceClasses/Abce/AbceCodeInterface.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on 2023-Jul-12
+
+This is a CodeInterface for the ABCE code.
+"""
+
+import os
+import re
+import warnings
+import sqlite3
+import pandas as pd
+
+from ravenframework.CodeInterfaceBaseClass import CodeInterfaceBase
+from ..Generic.GenericCodeInterface import GenericParser
+
+class Abce(CodeInterfaceBase):
+  """
+    This class is used to run the Abce code.
+  """
+
+  def generateCommand(self,inputFiles,executable,clargs=None, fargs=None, preExec=None):
+    """
+      See base class.  Collects all the clargs and the executable to produce the command-line call.
+      Returns tuple of commands and base file name for run.
+      Commands are a list of tuples, indicating parallel/serial and the execution command to use.
+      @ In, inputFiles, list, List of input files (length of the list depends on the number of inputs have been added in the Step is running this code)
+      @ In, executable, string, executable name with absolute path of ABCE/run.py
+      @ In, clargs, dict, optional, dictionary containing the command-line flags the user can specify in the input (e.g. under the node < Code >< clargstype =0 input0arg =0 i0extension =0 .inp0/ >< /Code >)
+      @ Out, returnCommand, tuple, tuple containing the generated command. returnCommand[0] is the command to run the code (string), returnCommand[1] is the name of the output root
```

Comment:
the input docs for `fargs` and `preExec` are missing from this docstring

Reply:
Neither are used.  How would you like that documented?

Reply:
Documented as described above.

Reply:
Changed to `ignored` to be consistent with rest of file.